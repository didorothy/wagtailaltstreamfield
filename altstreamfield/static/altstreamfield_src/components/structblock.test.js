import React from "react";
import { render, fireEvent, wait, waitForDomChange, waitForElementToBeRemoved, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import FormErrorContext from "../context/formerror";
import StructBlock from "./structblock";
import CharField from "./fields/charfield";

describe("StructBlock", () => {
    test("#minimal", () => {
        let manager = new FormErrorContext.FormErrorManager();
        let block = {
            id: '8d0f0232-07c8-11ea-8d71-362b9e155667',
            type: 'StructBlock',
            value: {
                'one': '',
            }
        };
        let fields = [{
            name: "one",
            field_type: CharField,
            args: {
                "name": "one",
                "label": "One",
                "required": true,
                "help_text": "Sample help text.",
                "strip": true,
                "min_length": null,
                "max_length": 255
            }
        }];
        let container = render(<FormErrorContext.Provider value={manager}>
            <StructBlock block={block} fields={fields} onChange={() => {}} />
        </FormErrorContext.Provider>);
        expect(container.container).toMatchSnapshot();
        cleanup();
    });

    test('handle change', () => {
        let manager = new FormErrorContext.FormErrorManager();
        let block = {
            id: '8d0f0232-07c8-11ea-8d71-362b9e155667',
            type: 'StructBlock',
            value: {
                'one': '',
            }
        };
        let fields = [{
            name: "one",
            field_type: CharField,
            args: {
                "name": "one",
                "label": "One",
                "required": true,
                "help_text": "Sample help text.",
                "strip": true,
                "min_length": null,
                "max_length": 255
            }
        }];
        function on_change(value) {
            block = value;
        }
        let container = render(<FormErrorContext.Provider value={manager}>
            <StructBlock block={block} fields={fields} onChange={on_change} />
        </FormErrorContext.Provider>);
        expect(manager.isValid()).toEqual(false);
        let input = container.container.getElementsByTagName('input')[0];
        fireEvent.change(input, {target: {value: 'test'}});
        container.rerender(<FormErrorContext.Provider value={manager}>
            <StructBlock block={block} fields={fields} onChange={on_change} />
        </FormErrorContext.Provider>)
        expect(manager.isValid()).toEqual(true);
        expect(block.value.one).toEqual('test');

    });
});