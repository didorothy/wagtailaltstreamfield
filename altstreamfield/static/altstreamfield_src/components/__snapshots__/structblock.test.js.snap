// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StructBlock #defaults 1`] = `
<div>
  <div
    class="struct-block"
    id="{this.props.block.id}"
  >
    <div
      class="field char_field required"
    >
      <label
        for="8d0f0232-07c8-11ea-8d71-362b9e155667one"
      >
        One
      </label>
      <div
        class="field-content"
      >
        <div
          class="input"
        >
          <input
            id="8d0f0232-07c8-11ea-8d71-362b9e155667one"
            placeholder="One"
            type="text"
            value="sailor"
          />
        </div>
        <p
          class="help"
        >
          Sample help text.
        </p>
        
      </div>
    </div>
  </div>
</div>
`;

exports[`StructBlock #minimal 1`] = `
<div>
  <div
    class="struct-block"
    id="{this.props.block.id}"
  >
    <div
      class="field char_field required error"
    >
      <label
        for="8d0f0232-07c8-11ea-8d71-362b9e155667one"
      >
        One
      </label>
      <div
        class="field-content"
      >
        <div
          class="input"
        >
          <input
            id="8d0f0232-07c8-11ea-8d71-362b9e155667one"
            placeholder="One"
            type="text"
            value=""
          />
        </div>
        <p
          class="help"
        >
          Sample help text.
        </p>
        <p
          class="error-message"
        >
          This field is required.
        </p>
      </div>
    </div>
  </div>
</div>
`;
